### PR TITLE
Frontend API client

### PR DESCRIPTION
**Pull request by KoalaSat**
This PR is the first step for https://github.com/Reckless-Satoshi/robosats/pull/170 , it creates an abstraction layer for the API requests (for this generic case `ApiWebClient`), so it can be easily replaced in future steps by other methods to communicate with the back-end (ex: `ApiNativeClient`).

I prefer to leave web-sockets as a totally different task and first validate this approach.

This change is huge and requires deep testing.

### CODE REVIEW

**Comment by KoalaSat** [comment]
I tested all endpoints and complete an entire order, everything seems to work, but would be great if someone double checks

File: frontend/src/components/UserGenPage.js [hunk 1]
```diff
@@ -63,27 +64,22 @@ class UserGenPage extends Component {
...
+        encrypted_private_key: key.encryptedPrivateKeyArmored,
+        unique_values: strength.uniqueValues,
+        counts: strength.counts,
+        length: token.length,
+        ref_code: refCode,
       };
     });
 
-    console.log(requestOptions);
+    console.log(requestBody);
```
Comment (Reckless-Satoshi) [inline frontend/src/components/UserGenPage.js:79]: I think we can use this change to get rid of this legacy `console.log()`

File: frontend/src/services/api/ApiWebClient/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    };
+    return fetch(path, requestOptions).then((response) => response.json())
+  }
+
+  public delete: (path: string) => Promise<object> = (path) => {
+    const requestOptions = {
+      method: 'DELETE',
+      headers: this.getHeaders(),
+    };
+    return fetch(path).then((response) => response.json())
```
Comment (Reckless-Satoshi) [inline frontend/src/services/api/ApiWebClient/index.ts:32]: `fetch(path)` is missing `requestOptions`, so this DELETE request behaves as GET. Robot deletion fails with 500 Error.

**Review by Reckless-Satoshi — COMMENTED** [review]
Looks neat an clean. I tested slightly but still need to fully test an end-to-end trade still to make sure we do not merge a broken frontend.

File: frontend/src/services/api/ApiWebClient/index.ts (hunk 2 above)
Comment (KoalaSat) [reply frontend/src/services/api/ApiWebClient/index.ts]: changed, good catch!

**Comment by Reckless-Satoshi** [comment]
Working as expected after a few full trades. Merging!

**Pull request merged**